framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -106,6 +106,65 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
 		return currentBlockSlot - finalizedBlockSlot > threeRounds;
 	}
 
+	/**
+	 * Request blocks from `fromID` ID to `toID` ID from an specific peer `peer` and applies them
+	 *
+	 * @param {object} peerId - The ID of the peer to target
+	 * @param {string} fromId - The starting block ID to fetch from
+	 * @param {string} toId - The ending block ID
+	 * @throws {ApplyPenaltyAndRestartError} - In case the peer hasn't returned any block
+	 */
+	async _requestAndApplyBlocksWithinIDs(peerId, fromId, toId) {
+		const maxFailedAttempts = 10; // TODO: Probably expose this to the configuration layer?
+		let failedAttempts = 0; // Failed attempt === the peer doesn't return any block or there is a network failure (no response or takes too long to answer)
+		let lastFetchedID = fromId;
+		let finished = false;
+
+		while (!finished && failedAttempts < maxFailedAttempts) {
+			const { data: blocks } = await this.channel.invoke(
```

Comment:
in case of this request fails or network error, it will throw and exit without restarting

Reply:
@shuse2 let's tackle this in another issue, as we're currently not handling network timeouts or request failures for both BSM and FCS, and because this is a direct cause for banning, we should spike it.